get error "invalid memory address or nil pointer dereference" when upgrade minio operator to v4.3.2 from v4.0.6

<!--- Provide a general summary of the issue in the Title above -->

## Expected Behavior
upgrade minio operator to v4.3.2 from v4.0.6 success, minio-operator and minio instance works well.

## Current Behavior
upgrade minio operator to v4.3.2 get an error in minio-operator, as follows:
```
I0211 09:21:26.105917       1 main.go:75] Starting MinIO Operator
I0211 09:21:26.539586       1 main.go:142] caBundle on CRD updated
I0211 09:21:26.543138       1 main-controller.go:252] Setting up event handlers
I0211 09:21:26.545697       1 leaderelection.go:243] attempting to acquire leader lease harbor-operator-ns/minio-operator-lock...
I0211 09:21:26.565271       1 leaderelection.go:253] successfully acquired lease harbor-operator-ns/minio-operator-lock
I0211 09:21:26.565613       1 main-controller.go:464] minio-operator-67ddbdd4d8-l64xv: I've become the leader
I0211 09:21:26.565707       1 main-controller.go:379] Waiting for API to start
I0211 09:21:26.618589       1 main-controller.go:361] Starting HTTPS API server
I0211 09:21:26.619046       1 main-controller.go:383] Starting Tenant controller
I0211 09:21:26.619095       1 main-controller.go:386] Waiting for informer caches to sync
I0211 09:21:26.619121       1 main-controller.go:391] Starting workers
I0211 09:21:26.727455       1 status.go:121] Hit conflict issue, getting latest version of tenant
I0211 09:21:31.573090       1 upgrades.go:84] Upgrading v4.2.9
E0211 09:21:31.573515       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 147 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x158c020, 0x254d3c0})
	k8s.io/apimachinery@v0.20.2/pkg/util/runtime/runtime.go:74 +0x85
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0008547f0})
	k8s.io/apimachinery@v0.20.2/pkg/util/runtime/runtime.go:48 +0x75
panic({0x158c020, 0x254d3c0})
	runtime/panic.go:1038 +0x215
github.com/minio/operator/pkg/controller/cluster.(*Controller).upgrade429(0xb, {0x1991278, 0xc000196000}, 0xc000c00018)
	github.com/minio/operator/pkg/controller/cluster/upgrades.go:271 +0x10d
github.com/minio/operator/pkg/controller/cluster.(*Controller).checkForUpgrades(0xc000490000, {0x1991278, 0xc000196000}, 0xc000416e00)
	github.com/minio/operator/pkg/controller/cluster/upgrades.go:85 +0x8c7
github.com/minio/operator/pkg/controller/cluster.(*Controller).syncHandler(0xc000490000, {0xc0002c8c30, 0x2c})
	github.com/minio/operator/pkg/controller/cluster/main-controller.go:630 +0x506
github.com/minio/operator/pkg/controller/cluster.(*Controller).processNextWorkItem.func1({0x1523ba0, 0xc0008547f0})
	github.com/minio/operator/pkg/controller/cluster/main-controller.go:557 +0x245
github.com/minio/operator/pkg/controller/cluster.(*Controller).processNextWorkItem(0xc000490000)
	github.com/minio/operator/pkg/controller/cluster/main-controller.go:569 +0x62
github.com/minio/operator/pkg/controller/cluster.(*Controller).runWorker(0x0)
	github.com/minio/operator/pkg/controller/cluster/main-controller.go:509 +0x70
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x7f1216d80c50)
	k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:155 +0x67
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x0, {0x19653e0, 0xc000665620}, 0x1, 0xc0001834a0)
	k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0, 0x3b9aca00, 0x0, 0x0, 0xc000808e90)
	k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(0xc000808fd0, 0x8782a6, 0xc000aca640)
	k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:90 +0x25
created by github.com/minio/operator/pkg/controller/cluster.(*Controller).Start.func1
	github.com/minio/operator/pkg/controller/cluster/main-controller.go:394 +0x2dd
E0211 09:21:31.573690       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
```
and the current Tenant.spec like:
```
  pools:
  - name: zone-harbor
    resources: {}
    securityContext:
      fsGroup: 10000
      runAsGroup: 10000
      runAsUser: 10000
    servers: 2
```
don't have pool.SecurityContext.RunAsNonRoot config.

## Possible Solution
https://github.com/minio/operator/blob/382365919b37e390b97eaf413fd2f62b3acad060/pkg/controller/cluster/upgrades.go#L271
Need to override the case when pool.SecurityContext.RunAsNonRoot is nil

## Steps to Reproduce (for bugs)
1. deploy minio operator with version v4.0.6.
2. deploy minio Tenant
3. upgrade minio operator to v4.3.2
ref issue: https://github.com/goharbor/harbor-operator/issues/833

Upgrade to v4.4.7 get an error as bellow:
```
0213 06:23:08.869400       1 main.go:71] Starting MinIO Operator
I0213 06:23:09.538020       1 main.go:150] caBundle on CRD updated
I0213 06:23:09.539564       1 main-controller.go:239] Setting up event handlers
I0213 06:23:09.540202       1 leaderelection.go:243] attempting to acquire leader lease harbor-operator-ns/minio-operator-lock...
I0213 06:23:09.586729       1 leaderelection.go:253] successfully acquired lease harbor-operator-ns/minio-operator-lock
I0213 06:23:09.587130       1 main-controller.go:450] minio-operator-6ffdb545bb-jrg2d: I've become the leader
I0213 06:23:09.588439       1 main-controller.go:365] Waiting for API to start
panic: tls: private key does not match public key

goroutine 112 [running]:
github.com/minio/operator/pkg/controller/cluster.(*Controller).generateTLSCert(0xc0004687e0)
	github.com/minio/operator/pkg/controller/cluster/operator.go:135 +0x525
github.com/minio/operator/pkg/controller/cluster.(*Controller).Start.func1.1()
	github.com/minio/operator/pkg/controller/cluster/main-controller.go:346 +0x12c
created by github.com/minio/operator/pkg/controller/cluster.(*Controller).Start.func1
	github.com/minio/operator/pkg/controller/cluster/main-controller.go:341 +0xb2
```

--- a/pkg/controller/cluster/upgrades.go
+++ b/pkg/controller/cluster/upgrades.go
@@ -266,12 +266,15 @@ func (c *Controller) upgrade429(ctx context.Context, tenant *miniov2.Tenant) (*m
 	if atLeastOnePoolInitialized {
 		for i, pool := range tenant.Spec.Pools {
 			// if they have a security context, and is having them run as root
-			if pool.SecurityContext != nil && !*pool.SecurityContext.RunAsNonRoot && *pool.SecurityContext.RunAsUser == 0 {
-				// Report the pool is now Legacy Security Context
-				tenant.Status.Pools[i].LegacySecurityContext = true
-				// push updates to status
-				if tenant, err = c.updatePoolStatus(ctx, tenant); err != nil {
-					klog.Errorf("'%s/%s' Error upgrading implicit tenant security context, MinIO may not start: %v", tenant.Namespace, tenant.Name, err)
+			scontext := pool.SecurityContext
+			if scontext != nil && scontext.RunAsNonRoot != nil && scontext.RunAsUser != nil {
+				if !*scontext.RunAsNonRoot && *scontext.RunAsUser == 0 {
+					// Report the pool is now Legacy Security Context
+					tenant.Status.Pools[i].LegacySecurityContext = true
+					// push updates to status
+					if tenant, err = c.updatePoolStatus(ctx, tenant); err != nil {
+						klog.Errorf("'%s/%s' Error upgrading implicit tenant security context, MinIO may not start: %v", tenant.Namespace, tenant.Name, err)
+					}
 				}
 			}
 		}

